File: BackendDotnet/Keeper.Context/Model/ProjectModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Keeper.Context.Model
{
    [Table("Projects")]
    public class ProjectModel
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        [StringLength(50)]
        public string Title { get; set; } = default!;
        public string? Description { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public bool IsDeleted { get; set; } = false;
        public Guid? TagId { get; set; }
        public virtual TagModel? Tag { get; set; }
        public Guid CreatedById { get; set; }
        public virtual UserModel CreatedBy { get; set; }
        public Guid? UpdatedById { get; set; }
        public virtual UserModel? UpdatedBy { get; set; }
    }
}

File: Frontend/src/Models/ProjectModel.ts
import type { IMail } from "./MailModel"

export interface IProject {
  id?: string
  title: string
  description: string
  createdOn?: Date
  createdBy?: string
  updatedOn?: Date
  updatedBy?: string
  tagTitle?: string
  mail?:IMail
  owner?:string[]
  contributers?:string[]
}
